### PR TITLE
[feature]strformat: add 2 'fmt' macros that use specified characters instead of '{}' 

### PR DESCRIPTION
**Pull request by demotomohiro**
I want to use ``strformat.fmt`` to put Nim variables in GLSL code that are embed as string literal in Nim code.
But GLSL code have many ``{`` and ``}`` characters and I don't want to escape them.
So I added 2 ``fmt`` macros that use specified characters instead of ``{`` and ``}``.

### CODE REVIEW

**Comment by krux02** [comment]
I like the idea and I also see the problem with `{}` as they are used in many programming languages. I have a few remarks though. 

I would leave out the single open/close char variant ``macro fmt*(pattern: string; openCloseChar: char): untyped``, because I don't think it is that useful, and it can be left out without reducing the flexibility. I think the correct way to use these is to define a local forwarding template that sets both the open and the close char, and then call fmt only with this template in the entire project:

```nim
template myFmt*(arg: string): string = fmt(arg, '<', '>')
```

The second remark I have is, the char argument only works for ASCII chars. When I would like to have some unicode delimiters where I know they won't collide, for example ``fmt("...{ ⟨x⟩[i] < ⟨y⟩[i]; foo('x') } ...", "⟨", "⟩")``, I can't do that with char. So can you make the delimiter a string?

The last remark that I have is unrelated to this PR. You are talking about injecting Nim variables in GLSL. This is something that I have already done successfully in this project here: https://github.com/krux02/opengl-sandbox. Maybe we should discuss if it is a good idea to collaborate.

**Comment by Araq** [comment]
> So can you make the delimiter a string?

That would make the implementation annoying to follow IMO, single chars are fine.

**Comment by demotomohiro** [comment]
@krux, @Araq
Thank you for your feedback!
I removed single open/close char variant ``fmt`` macro.

I changed the delimiter to a string and wrote the test but it is not commited.
I changed code like replacing ``if f[i] == '{'`` to ``if f.continuesWith(openStr, i)`` and corresponding ``inc i`` to ``inc i, openStr.len``.
It might a bit annoying compared to original code but I don't think it is hard to following.

What if ``fmt`` is used to a language that use most of ascii characters?
(I think Nim is one of such languages)
If the delimiter is limited to a ascii character, I have to choose a least used character and escape all these characters in an input string.
If I can use a string as delimiter, I can use multiple ascii characters or an unicode character as delimiter and I don't need to escape characters in an input string.
